adr/0010-release-process.md

```diff
@@ -0,0 +1,22 @@
+# Release Process
+
+Date: 2018-12-03
+
+## Status
+
+Draft
+
+## Context
+
+eLife has some projects that are published as packages, e.g. for Python projects on pypi.
+
+Specific details may vary from project to project. For example one project may just pushed to pypi, another to Docker Hub, or even both. Some artefacts may just be pushed for releases, others on commit.
+
+## Decision
+
+A new release will be indicated by a new git tag.
+Jenkins will trigger on a new tag to, beyond building and running tests, also push the release out (e.g. pypi, Docker Hub, etc.).
```

Comment:
investigating this bit, which I have never used in anger

Reply:
Turned on build of tags not older than 7 days, at https://alfred.elifesciences.org/job/pull-requests-projects/computation/console

Reply:
Tag ignored by the scan, still located at https://alfred.elifesciences.org/job/pull-requests-projects/computation/console

Reply:
Also had to add `Discover tags` to the branch sources.

Reply:
Discovered, but didn't build until I manually triggered it: https://alfred.elifesciences.org/job/pull-requests-projects/job/elife-jenkins-workflow-libs/view/tags/job/0.0.1/

Reply:
Seems these jobs have a `TAG_NAME=0.0.1` environment variable we can use to check.

Reply:
https://github.com/elifesciences/elife-jenkins-workflow-libs/pull/14

Reply:
https://alfred.elifesciences.org/job/pull-requests-projects/job/elife-jenkins-workflow-libs/view/tags/job/0.0.2/ correctly triggered after being pushed (the previous one was already existing so only discovered but not built).

Reply:
Also https://github.com/elifesciences/elife-jenkins-workflow-libs/pull/15